src/Workspaces/Core/Portable/Rename/ConflictEngine/DeclarationConflictHelpers.cs

```diff
@@ -37,20 +37,23 @@ public static IEnumerable<Location> GetMembersWithConflictingSignatures(IMethodS
                     if (!(conflictingSymbol.PartialDefinitionPart != null && conflictingSymbol.PartialDefinitionPart == renamedMethod) &&
                         !(conflictingSymbol.PartialImplementationPart != null && conflictingSymbol.PartialImplementationPart == renamedMethod))
                     {
-                        foreach (var location in conflictingSymbol.Locations)
-                        {
-                            yield return location;
-                        }
+                        builder.AddRange(conflictingSymbol.Locations);
                     }
                 }
             }
+
+            return builder.ToImmutableAndFree();
```

Comment:
@heejaechang @amcasey it does use a singleton instance.  `ToImmutableAndFree` resolves down to `ToImmutable` which does 0 optimizations

https://github.com/dotnet/corefx/blob/master/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray%601+Builder.cs#L175